src/components/PokemonsPage.tsx

```diff
@@ -0,0 +1,84 @@
+import { Component } from 'react';
+import Header from './Header';
+import Main from './Main';
+import { PokemonsPageState } from '../interface/PokemonsPageState';
+import { PokemonsPageProps } from '../interface/PokemonsPageProps';
+import { PokemonResponseItem } from '../interface/PokemonResponseItem';
+import Loader from './Loader';
+
+export default class PokemonsPage extends Component<
+  PokemonsPageProps,
+  PokemonsPageState
+> {
+  constructor(props: PokemonsPageProps) {
+    super(props);
+    this.state = {
+      isLoading: true,
+      pokemonData: [],
+    };
+    this.getPokemonsListData = this.getPokemonsListData.bind(this);
+    this.getPokemonData = this.getPokemonData.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  componentDidMount() {
+    const searchValue = localStorage.getItem('pokemonName');
+    if (!searchValue) this.getPokemonsListData();
+    else this.getPokemonData(searchValue);
```

Comment:
Better not to skip brackets if you have else.
```suggestion
    if (!searchValue) {
         this.getPokemonsListData();
    } else {
         this.getPokemonData(searchValue);
    }
```